### DISCUSSION TITLE
Scheduler stop importing DAG after failed task

### DISCUSSION THREAD

**Discussion by fokmess** [apache/airflow#32611] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

- Airflow version: **2.4.3**

### What happened

I have bug:
When task fail with this:

> [2023-07-13, 21:11:04 UTC] {logging_mixin.py:137} INFO - <generator object SQLDatabase._query_iterator at 0x7f5412c4d740>
> [2023-07-13, 21:15:34 UTC] {local_task_job.py:159} INFO - Task exited with return code Negsignal.SIGKILL
> [2023-07-13, 21:15:35 UTC] {taskinstance.py:2623} INFO - 0 downstream tasks scheduled from follow-on schedule check

The scheduler stop importing DAG and start saying **DAG seems to be missing**

In scheduler log when it attempt to parse dag it write that retrieved this dag:
> [2023-07-14T01:59:58.586+0000] {processor.py:766} INFO - DAG(s) dict_keys(['MainLoader']) retrieved from /home/listopadov/airflow/dags/EVRAZ_DAGS_GENERATOR.py

But below it write:
> [2023-07-14T06:31:49.645+0000] {logging_mixin.py:137} INFO - [2023-07-14T06:31:49.644+0000] {taskinstance.py:1853} ERROR - {'DAG Id': 'MainLoader', 'Task Id': 'LOADING_DATA.Load_Tran.insert_data', 'Run Id': 'scheduled__2023-07-12T21:00:00+00:00', 'Hostname': 'localhost'}

Scheduler write it to log until restart, after this one the dag importing correctly.

### How to reproduce

I think it happened whet Linux doesn't how enough RAM for task and marked failed, but something goes wrong

### Operating System

Ubuntu 20.04.4 LTS

### Versions of Apache Airflow Providers

apache-airflow==2.4.3

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
Yep. If there is memory or other reasons why resources are missing, this is fairly reasonable to expect that any system will be unstable and until you restart it there is no way it can recover on its own - this is precisely the moment your monitoring should inform you and people who are managing the platform should take actions (increasing resources, restarting everything).

Converting it into discussion in case more discussion is needed, but there is no action from Airflow we could (and would like to) do on this one.
